### ISSUE TITLE
Ошибка в схеме

### ISSUE THREAD

**Issue by nomad605dis** [Shatki/pyTroykaIMU#1]
Дмитрий, у вас на схеме подключения вместо IMU-сенсора, нарисован «Расширитель GPIO-портов».

**Comment by Shatki** [comment]
Да, это я для образца) Перерисую Еще не нашел всех библиотек для  Fritzing)
Был бы признателен если поделитесь. Нашел не для всех Тройка модулей   

>Четверг, 15 марта 2018, 13:10 +03:00 от Igor <notifications@github.com>:
>
>Дмитрий, у вас на схеме подключения вместо IMU-сенсора, нарисован «Расширитель GPIO-портов».
>—
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly,  view it on GitHub , or  mute the thread .


-- 
Дмитрий Селивёрстов

**Comment by nomad605dis** [comment]
Все SVG которые у нас есть мы выложили на Github. Рисуем их по мере необходимости и сразу же добавляем в репозиторий.

**Comment by nomad605dis** [comment]
Дмитрий, проверьте пожалуйста вашу почту, я вам пару вопросов написал.

**Comment by Shatki** [comment]
Игорь, сегодня проверю дома и отвечу. Спасибо!

**Comment by nomad605dis** [comment]
Хорошо, буду ждать.

**Comment by Shatki** [comment]
Игорь, исправил описание, и ответил в письме на вопросы. Обновите библиотеку. Если будут замечания, с удовольствием доработаю.

**Comment by nomad605dis** [comment]
Добрый день, Дмитрий!

Всё заработало. Теперь остался вопрос насчёт калибровки магнитометра.

В нашем примере мы каждый магнитометр калибруем, получаем калибровочные
коэффициенты, а потом уже используем.

Подскажите как это у вас реализовано.

15 марта 2018 г., 20:21 пользователь Дмитрий <notifications@github.com>
написал:

**Comment by Shatki** [comment]
Добрый день, Игорь! 
Для калибровки я использовал скрипт pyTroykaIMU/igrf12py/lis3mdl_calibration.py. После его запуска нужно немного покрутить в руках модуль  для получения файла lis3mdl_calibration.txt . После этого в программе magneto.exe  рассчитывались калибровочные коэффициенты, на основании этого файла, которые можно передать через аналогичную в Вашей реализации библиотеки функцию. Делал года полтора назад. Помню, меня не устроило использование сторонних утилит и я пытался сделать это в реальном времени, но что-то отвлекся на другие проекты. Опробую всю эту методику заново, и напишу подробное руководство для этой процедуры.

**Comment by nomad605dis** [comment]
Да и есть проблема с выводами азимута и напряжённости магнитного поля.

Мы решили сделать по одному примеру для каждого модуля: акселерометра,
гироскопа, компаса и барометра.

Так вот на компасе возникли сложности:
1. Какой процесс калибровки.
2. Даже если не калибровать, векторы напряжённости не меняются, а если
выводить азимут программа вообще зависает.

Сам код:
from pytroykaimu import TroykaIMU
import time

imu = TroykaIMU()

while True:
    dataCompass = imu.magnetometer.read_azimut()
# dataCompass = imu.magnetometer.read_gauss_xyz()
    if dataCompass:
        print(dataCompass)
        time.sleep(0.1)



16 марта 2018 г., 13:48 пользователь Игорь Дементьев <igor@amperka.ru>
написал:

**Comment by nomad605dis** [comment]
Только увидел ваше предыдущее письмо. Вообщем тогда ждём результатов от
компасса. Акселерометр и Гироскоп всё ок. Сейчас протестируем барометр.

Заранее спасибо)

16 марта 2018 г., 14:16 пользователь Игорь Дементьев <igor@amperka.ru>
написал:

**Comment by Shatki** [comment]
Игорь обновите библиотеку целиком или только для магнитометра, и попробуйте. Утром сделал небольшие доработки

**Comment by nomad605dis** [comment]
Обновили либу. Данные с компаса датчик отдаёт. Но вот при вызове азимута программа зависает.
imu.magnetometer.read_azimut()

**Comment by Shatki** [comment]
Принято, сегодня проверю, спасибо.

**Comment by nomad605dis** [comment]
Добрый день, Дмитрий!

Ну как там ваши успехи?

**Comment by Shatki** [comment]
Игорь, привет! Скоро добавлю в библиотеку все необходимое. Я переделываю максимально близко к вашей Ардуиновской библиотеке, чтоб было проще использовать. Нужно 2-3 дня, думаю) 
>Вторник, 20 марта 2018, 12:43 +03:00 от Igor <notifications@github.com>:
>
>Добрый день, Дмитрий!
>Ну как там ваши успехи?
>—
>You are receiving this because you commented.
>Reply to this email directly,  view it on GitHub , or  mute the thread .


-- 
Дмитрий Селивёрстов

**Comment by nomad605dis** [comment]
Хорошо. Без проблем. Тогда сообщите потом о результатах.

Заранее спасибо.

**Comment by Shatki** [comment]
Хорошо Игорь, в ближайшее время напишу.


>Вторник, 20 марта 2018, 12:57 +03:00 от Igor <notifications@github.com>:
>
>Хорошо. Без проблем. Тогда сообщите потом о результатах.
>Заранее спасибо.
>—
>You are receiving this because you commented.
>Reply to this email directly,  view it on GitHub , or  mute the thread .


-- 
Дмитрий Селивёрстов

**Comment by Shatki** [comment]
Игорь, привет! доработал библиотеку, Азимут работает, с калибровкой еще лучше работает. Сегодня опишу процедуру калибровки. Можно обновить либу и пробовать. Скоро еще тестовый скрипт с управлением 3D самолетом добавлю.
